### PR TITLE
Trustregion cfgblk

### PR DESCRIPTION
**Pull request by dmvigi**
## Summary/Motivation:
Specify parameters using PyUtilib ConfigBlocks.

## Changes proposed in this PR:
- ConfigBlock parameters now define default values
- Changes to default values can be specified at initialization of solver or when the solve function is called. 

NOTE: There is a bug that throws errors on subsequent calls to the solve function. To show that the ConfigBlock code is performing as expected, I added six separate tests in the test directory.  I will collapse these tests into a single test file once the bug is fixed.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by blnicho** [comment]
@dmvigi I took a quick glance at the tests that are failing and it looks like there is an outdate import statement. `config.py` was moved from `pyomo.util` to `pyomo.common`. You might have to wipe all `.pyc` files in your local clone to recreate this import error. Could you update the imports before I do a more thorough review? Also, make sure to update your Pyutilib clone as @jsiirola recently merged a fix for the bug you found.

**Comment by dmvigi** [comment]
I fixed earlier errors that prevented clean tests. The bug that was throwing errors on subsequent calls to the solve function must have been a result of the outdated import command. I collapsed these tests into a single test file.

**Comment by dmvigi** [comment]
@jsiirola or @blnicho I am not sure what to do to fix the documentation error--help?

**Comment by jsiirola** [comment]
@dmvigi: I don't think it was a documentation issue.  Looking at Travis, only the "slim" build was failing.  "slim" is a build that only installs the minimal Pyomo dependencies - which does not include numpy.  The TRF/py code has a hard dependency on numpy, so we need to skip the tests when numpy is not present.  I just pushed that change (eaa1d75) and we will see if the tests clear up.

File: pyomo/contrib/trustregion/plugins.py [hunk 1]
```diff
@@ -29,13 +39,227 @@ class TrustRegionSolver(OptSolver):
...
+        # set persistent config options
+        self.config = self.CONFIG(kwds)
+        #print(self.config.value())
+
         #
         # Call base class constructor
         #
+ 
         kwds['type'] = 'trustregion'
+        kwds['solver'] = 'ipopt'
```
[the remark this answers is not in the document]
Comment (jsiirola) [reply pyomo/contrib/trustregion/plugins.py]: Here's another idea.  I think what you want to do is pass a "clean" kwds in to the base class.  I don't remember what "type" is -- I think it is needed by the OptSolver base class, but I don't know what for.  I think we can remove `type` from the ConfigBlock (it is not something a user would change).
```
# Pop off any keywords this ConfigBlock doesn't recognize and put them into a temporary dict to pass on to the base class
tmp_kwds = {'type': kwds.pop('type','trustregion')}
self.config = self.CONFIG(kwds)

# Add any options needed by the base class o the temporary dict *from* the local ConfigBlock (this way we will pick up any default values from unspecified options)
tmp_kwds['solver'] = self.config.solver

# finally, initialize the base class
_OptSolver.__init__(**tmp_kwds)
```

**Pull request merged**